example/src/main/java/com/stripe/example/activity/CustomerSessionActivity.java

```diff
@@ -68,19 +73,44 @@ public void onClick(View view) {
     }
 
     private void launchWithCustomer() {
-        new PaymentMethodsActivityStarter(this).startForResult(REQUEST_CODE_SELECT_SOURCE);
+        final Intent paymentMethodsIntent = new PaymentMethodsActivityStarter(this).newIntent()
+                .putExtra(PaymentMethodsActivity.EXTRA_SHOW_GOOGLE_PAY, true);
+        this.startActivityForResult(paymentMethodsIntent, REQUEST_CODE_SELECT_SOURCE);
+
+        // new PaymentMethodsActivityStarter(this).startForResult(REQUEST_CODE_SELECT_SOURCE);
+    }
+
+    @Nullable
+    private String readPaymentObjectType(String selectedPayment) {
+        // This should be available through a higher level static
+        JSONObject jsonObject;
+        try {
+            jsonObject = new JSONObject(selectedPayment);
+        } catch (JSONException ignored) {
+            jsonObject = null;
+        }
+        if(jsonObject == null)
+            return null;
+
+        return jsonObject.optString(StripePaymentSource.FIELD_OBJECT);
```

Comment:
why do you need to put the whole JSON object in the extra if all you need is a single field?

Reply:
donnow man, but that's not up to me. that's how the dialog works. 
it returns a card object normally. which has the whole card data. i don't know why the card was returned...especially considering that's what's gonna be charged is the customer...but that's how it's done, also in the iOS implementation...so i didn't argue. and i guess it's useful for debugging or whatnot.

What i did was to add a field to this object which states the type of returned element. Inspired by the iOS implementation (and it also makes sense), this will tell you whether you can now followup with charging the _customer_ OR need to go ahead with the native method (here being google pay), get a stripe token from there and charge _the token_.

I guess I could replace it with another extra string extra, but it feels more natural to just return it with the existing dialog result variable.

Reply:
you know what...yeah. how bout i change this to an extra param. a bit overdoing it.

Reply:
figured i'll "add" rather than change. For clients that always read the result into json (like well...react native ones, which is my end goal).

Reply:
i'll add another extra to mention the type, and use it instead